### PR TITLE
Add a debug metric command

### PR DESCRIPTION
**Pull request by jgustie**
This adds a hidden command that generates a Prometheus unit test based on the rendered queries from an experiment's metrics (non-Prometheus metrics are just put in a YAML comment).

### CODE REVIEW

File: redskyctl/internal/commands/debug/metric.go [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+	_ = cmd.MarkFlagFilename("filename", "yml", "yaml")
+	_ = cmd.MarkFlagRequired("filename")
+
+	return cmd
+}
+
+func (o *MetricQueryOptions) Debug() error {
+	// Read the experiment
+	r, err := o.IOStreams.OpenFile(o.Filename)
```
Comment (bradbeam) [inline redskyctl/internal/commands/debug/metric.go:76]: possible enhancement for later would be to fetch this from the running cluster

File: redskyctl/internal/commands/debug/metric.go [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+	t.Status.CompletionTime = &completionTime
+	return nil
+}
+
+func parseTime(input string, defaultTime time.Time) (metav1.Time, error) {
+	if input == "" {
+		return metav1.NewTime(defaultTime), nil
+	}
+
+	ss := regexp.MustCompile(`^([0-9]+)(?:\.([0-9]+))?$`).FindStringSubmatch(input)
```
Comment (bradbeam) [inline redskyctl/internal/commands/debug/metric.go:172]: Can you add a comment that this catches unixtime?

Comment (bradbeam) [inline redskyctl/internal/commands/debug/metric.go:172]: Does this allow for a non-nano unix time? as in if I do `date +%s` can I feed that in directly?

Comment (bradbeam) [inline redskyctl/internal/commands/debug/metric.go:172]: and can I say that I'm just as frustrated that there is not a `time.Unix/time.UnixNano` constant for the format?

File: redskyctl/internal/commands/debug/metric.go [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+				{Kind: yaml.ScalarNode, Value: "promql_expr_test", HeadComment: "\n"},
+				{Kind: yaml.SequenceNode},
+			},
+		}),
+	)
+}
+
+func (p *promTest) addPromqlExprTest(t *redskyv1beta1.Trial, m *redskyv1beta1.Metric, q string) error {
+	return p.document.PipeE(
```
Comment (bradbeam) [inline redskyctl/internal/commands/debug/metric.go:274]: I think this is good. 
We may want to do some bogus/fake data insertion which we could cheese by using the arithmetic-like expression ( I don't know the actual format, but im thinking something like `1x<duration in seconds>` so we could have a hopefully easily calculable value to validate against. 
 
```
  - input_series:
      - series: 'go_goroutines{job="prometheus", instance="localhost:9090"}'
        values: '10+10x2 30+20x5' # 10 20 30 30 50 70 90 110 130
      - series: 'go_goroutines{job="node_exporter", instance="localhost:9100"}'
        values: '10+10x7 10+30x4' # 10 20 30 40 50 60 70 80 10 40 70 100 130
```

File: redskyctl/internal/commands/debug/metric.go [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+		// Render the query (record errors as a comment)
+		q, _, err := eng.RenderMetricQueries(m, t, target)
+		if err != nil {
+			pt.addErrorComment(m, err)
+			continue
+		}
+
+		// Not Prometheus, just add a comment to the header
+		if m.Type != redskyv1beta1.MetricPrometheus {
```
Comment (bradbeam) [inline redskyctl/internal/commands/debug/metric.go:130]: hadn't thought about this, good catch.

**Review by bradbeam — COMMENTED** [review]
looks good; just some questions/comments around time. because time is hard.

File: redskyctl/internal/commands/debug/metric.go (hunk 1 above)
Comment (jgustie) [reply redskyctl/internal/commands/debug/metric.go:76]: Process substitution all the way. I was playing around with this using mostly `redskyctl debug metric -f <(redskyctl gen exp -f app.yaml)` but you can use `kubectl get exp foo -o yaml` also.

File: redskyctl/internal/commands/debug/metric.go (hunk 2 above)
Comment (jgustie) [reply redskyctl/internal/commands/debug/metric.go:172]: I was equally surprised.

**Review by bradbeam — APPROVED** [review]

**Pull request merged**